dags/aps/parser.py

```diff
@@ -46,53 +50,53 @@ def __init__(self) -> None:
             NestedValueExtractor("journal_issue", json_path="issue.number"),
             NestedValueExtractor("journal_volume", json_path="volume.number"),
             NestedValueExtractor(
-                "journal_year", json_path="date", extra_function=lambda x: int(x[:4])
+                "journal_year",
+                json_path="date",
+                extra_function=lambda x: int(x[:4] if (len(x) >= 4) else 0000),
             ),
             NestedValueExtractor("date_published", json_path="date"),
             NestedValueExtractor(
                 "copyright_holder",
                 json_path="rights.copyrightHolders",
-                extra_function=lambda x: x[0]["name"],
+                extra_function=lambda x: x[0]["name"] if len(x) >= 1 else "",
```

Comment:
```suggestion
                extra_function=lambda x: x[0]["name"] if len(x) >= 0 else "",
```
?

Reply:
I'm the suggestion is wrong. For example, we have:
```
x=[{'name':  'Copy Holder Name'}]
```
the length of this array is 1, so we can take the first item (x[0]) name.
If
```
x=[]
```
the length of this array is 0, and when we will try to take the value of x[0]['name'] it will give us an error of:
```
IndexError: list index out of range
```